File: packages/localization/ios/Plugin/Classes/Results/GetLocalesResult.swift
import Foundation
import Capacitor

@objc public class GetLocalesResult: NSObject, Result {
    let locales: [LocaleResult]

    init(locales: [LocaleResult]) {
        self.locales = locales
    }

    @objc public func toJSObject() -> AnyObject {
        let localesArray: [JSValue] = locales.compactMap { $0.toJSObject() as? JSObject }
        var result = JSObject()
        result["locales"] = localesArray
        return result as AnyObject
    }
}

File: packages/localization/android/src/main/java/io/capawesome/capacitorjs/plugins/localization/classes/results/GetLocalesResult.java
package io.capawesome.capacitorjs.plugins.localization.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSArray;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.localization.interfaces.Result;
import java.util.List;

public class GetLocalesResult implements Result {

    @NonNull
    private final List<LocaleResult> locales;

    public GetLocalesResult(@NonNull List<LocaleResult> locales) {
        this.locales = locales;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSArray localesArray = new JSArray();
        for (LocaleResult locale : locales) {
            localesArray.put(locale.toJSObject());
        }
        JSObject result = new JSObject();
        result.put("locales", localesArray);
        return result;
    }
}
